pkg/snapstore/gcs_snapstore.go

```diff
@@ -167,6 +167,24 @@ func (s *GCSSnapStore) Save(snap brtypes.Snapshot, rc io.ReadCloser) error {
 		return fmt.Errorf("failed uploading composite object for snapshot with error: %v", err)
 	}
 	logrus.Info("Composite object uploaded successfully.")
+
+	// New context to be used for deleting chunks. Timeout same as upload chunk timeout.
+	chunkDeleteCtx, chunkDeleteCancel := context.WithTimeout(context.TODO(), chunkUploadTimeout)
+	defer chunkDeleteCancel()
+	// Delete the chunks right after uploading the snapshot to bucket.
+	// To be done once the composite object is successfully uploaded.
+	logrus.Infof("Started deleting the chunk objects from bucket: GCS")
+	errList := deleteChunksGCS(chunkDeleteCtx, snap, prefix, noOfChunks, bh)
+
+	// Log the errors occured while deleting the chunks
+	for _, errLog := range errList {
+		fmt.Println(errLog)
+	}
+
+	if len(errList) > 0 {
+		return fmt.Errorf("failed deleting chunks in GCS for snapshot: %v", snap.SnapName)
+	}
+	logrus.Infof("Successfully deleted %d chunks from GCS bucket", noOfChunks)
```

Comment:
Currently, we're deleting chunks one by one, which slows down the deletion process. We should consider using a parallel deletion approach, similar to our upload method with `collectChunkUploadError`, to speed things up.

Reply:
Thanks sesha. Addressed the review comment in latest commit. Closing this